explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,405 @@
+[variants and variant-key](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+attributes to support content negotiation for recursive prefetch.
+
+- In **outer** HTTP response of article.html.sxg from feed.example:
+    ```
+    Link: <https://feed.example/publisher.example/image_jpeg.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          anchor="https://publisher.example/image";
+    Link: <https://feed.example/publisher.example/image_webp.sxg>;
+          rel="alternate";
+          type="application/signed-exchange;v=b3";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          anchor="https://publisher.example/image";
+    ```
+- In **inner** response header of article.html.sxg:
+    ```
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/jpeg";
+          header-integrity="sha256-AAA"
+    Link: <https://publisher.example/image>;
+          rel="allowed-alt-sxg";
+          variants-05="accept;image/jpeg;image/webp";
+          variant-key-05="image/webp";
+          header-integrity="sha256-BBB"
+    Link: <https://publisher.example/image>; rel=preload; as=image;
+    ```
+If a UA supports WebP, the UA must prefetch **image_webp.sxg** holding a WebP
+image. Otherwise the UA must prefetch **image_jpeg.sxg** holding a JPEG image.
+
+## Security and Privacy Considerations
+- The publishers can know whether the referrer page has prefetched the signed
+exchange subresources or not by checking the resource timing information. But
+this only exposes 1 bit information (= yes or no) because UAs can use the cached
+signed exchange only if the required signed exchanges are all available.
+- To prevent tracking (user ID transfer), if the aggregator failed to prefetch a
+subresource that the main resource preloads, the UA must drop all of the
+subresource prefetches. If the aggregator prefetches a superset of the preloaded
+subresources, the UA must drop the ones that weren't preloaded.
+- The UA can use the prefetched signed exchange subresources, only when they
+were prefetched in the rimmediate eferrer page. This is intended to avoid
+leaking the prefetching state to succeeding pages.
+- The UA check the header integrity value, so the distributor of the subresource
+signed exchange can’t inject arbitrary resources to the publisher’s page. This
+prevents distributors from sending tracking IDs to the publisher’s page.
+- The UA needs to check that the signed
+[request URL](https://wicg.github.io/webpackage/loading.html#exchange-request-url)
+matches the preload link and not just that the header-integrity value matches,
+since the header-integrity hash doesn't cover the request URL.
+- The UA must fetch the signed exchange subresource
```

Comment:
It's more that the UA's decision of whether to fetch signed exchange subresources MUST NOT depend on whether the HTTP cache already contains the original subresource. The UA is free to avoid fetching them for other reasons.